sdk/src/wallet/migration/mod.rs

```diff
@@ -3,21 +3,52 @@
 
 mod migrate_0;
 
+use std::collections::HashMap;
+
+use anymap::Map;
 use async_trait::async_trait;
+use once_cell::sync::Lazy;
 use serde::{Deserialize, Serialize};
 
-use crate::wallet::Result;
-
-pub type LatestMigration = migrate_0::Migrate;
+use crate::{
+    client::storage::StorageAdapter,
+    wallet::{Error, Result},
+};
 
 pub(crate) const MIGRATION_VERSION_KEY: &str = "migration-version";
 
-/// The list of migrations, in order.
-const MIGRATIONS: &[&'static dyn DynMigration] = &[
-    // In order to add a new migration, change the `LatestMigration` type above and add an entry at the bottom of this
-    // list.
-    &migrate_0::Migrate,
-];
+#[cfg(feature = "stronghold")]
+struct LatestBackupMigration(MigrationVersion);
+
+static MIGRATIONS: Lazy<Map<dyn anymap::any::Any + Send + Sync>> = Lazy::new(|| {
+    let mut migrations = Map::new();
+    #[cfg(feature = "storage")]
+    {
+        use super::storage::Storage;
+        const STORAGE_MIGRATIONS: [(Option<usize>, &'static dyn DynMigration<Storage>); 1] = [
+            // In order to add a new storage migration, add an entry at the bottom of this list
+            // and change the list length above.
+            // The entry should be in the form of a key-value pair, from previous migration to next.
+            // i.e. (Some(migrate_<N>::Migrate::ID), &migrate_<N+1>::Migrate)
+            (None, &migrate_0::Migrate),
+        ];
+        migrations.insert(std::collections::HashMap::from(STORAGE_MIGRATIONS));
+    }
+    #[cfg(feature = "stronghold")]
+    {
+        use crate::client::stronghold::StrongholdAdapter;
+        const BACKUP_MIGRATIONS: [(Option<usize>, &'static dyn DynMigration<StrongholdAdapter>); 1] = [
+            // In order to add a new backup migration, and add an entry at the bottom of this list
+            // and change the list length above.
+            // The entry should be in the form of a key-value pair, from previous migration to next.
+            // i.e. (Some(migrate_<N>::Migrate::ID), &migrate_<N+1>::Migrate)
+            (None, &migrate_0::Migrate),
+        ];
+        migrations.insert(LatestBackupMigration(BACKUP_MIGRATIONS.last().unwrap().1.version()));
```

Comment:
Why do we need this extra for the backup but not for the storage?

Reply:
Because the storage always sets this value when created, but the backup cannot do that. Instead, backup sets the value on read or write.